src/js_realm.hpp

```diff
@@ -322,14 +337,17 @@ void Realm<T>::constructor(ContextType ctx, ObjectType this_object, size_t argc,
     ensure_directory_exists_for_file(config.path);
 
     SharedRealm realm = realm::Realm::get_shared_realm(config);
-    auto delegate = new RealmDelegate<T>(realm, Context<T>::get_global_context(ctx));
-
-    if (!realm->m_binding_context) {
-        realm->m_binding_context.reset(delegate);
+    
+    // Fix for datetime -> timestamp conversion
+    if (realm->config().upgrade_initial_version != realm->config().upgrade_final_version &&
+        realm->config().upgrade_initial_version < 5) {
+        assert(0);
```

Comment:
Maybe this needs a comment, but what is so bad about this condition that it needs to crash?